### PR TITLE
Feature 71 - useWindowDimensions

### LINKED ISSUE #71: Feature - useWindowDimensions [Dev]
## Checklist ✅

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/platzily/platzily-ui/issues) of this repository and believe that this is not a duplicate.

## Summary 💡

`useWindowDimensions` is a hook exposed from styling api, to let the user manage mediaqueries it is inspired from https://github.com/Sphinx-Society/kaizen-web-client/blob/master/src/client/hooks/useWindowDimensions/useWindowDimensions.js

It will be used like this:

```jsx
import { useWindowDimensions, useTheme } from '@platzily-ui/styling'

const MyComponent = () => {
  const { width } = useWindowDimensions();
  const theme = useTheme()
  const isMobile = width < theme.breakpoints.mobile + 1;

  if (isMobile) {
    return (
      <MobileUI/>
    )
  }

  return <NotMobileUI/>
}
```

It could be combined with `createStyleSheet` too

```jsx
import { useWindowDimensions, createStyleSheet, useTheme } from '@platzily-ui/styling'

const baseStyles = {}
const mobileStyles = {}
const notMobileStyles = {}

const useStyleSheet = createStyleSheet((_, props) => ({
  root: {
    ...baseStyles,
    ...(props.isMobile ? { ...mobileStyles } : { ...notMobileStyles })
  }
}))

const MyComponent = () => {
  const { width } = useWindowDimensions();
  const theme = useTheme()
  const isMobile = width < theme.breakpoints.mobile + 1;

  const { classes } = useStyleSheet({ isMobile })

  return <MobileAndNotMobileUI className={classes.root}/>
}

```

## Requirements 🌈

The hook must be tested

## Motivation 🔦

A way to provide an user friendly way to manage mediaqueries

### PR DESCRIPTION
**Pull request by omarefg**
## Checklist ✅

- [x] I have followed the [contributor guideline](https://github.com/platzily/platzily-ui/blob/master/CONTRIBUTING.md)
- [x] Tests are included
- [x] Documentation is changed or added
- [x] Related issue has been created
- [x] Commits messages follows [commit guideline](https://github.com/platzily/platzily-ui/blob/master/CONTRIBUTING.md/#Commits)

resolves #71

### CODE REVIEW

**Comment by kevfarid** [comment]
Please, add name of issue to PR. Example: Feature 71 - useWindowDimensions

**Comment by omarefg** [comment]
> Please, add name of issue to PR. Example: Feature 71 - useWindowDimensions

Done :heavy_check_mark:

File: .eslintrc.js [hunk 1]
```diff
@@ -81,7 +81,8 @@ module.exports = {
     'react/no-invalid-html-attribute': 'off',
     'react/no-unused-class-component-methods': 'off',
     'jsx-a11y/alt-text': 'off',
-    'react/function-component-definition': 'off'
+    'react/function-component-definition': 'off',
+    'no-restricted-exports': 'off'
```
Comment (KevFarid) [inline .eslintrc.js:85]: Can you use overrides to turn off this rule?

**Review by kevfarid — APPROVED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (kevfarid) [inline .eslintrc.js]: Can you use overrides to turn off this rule?

**Review by kevfarid — CHANGES_REQUESTED** [review]

**Review by kevfarid — APPROVED** [review]

**Pull request merged**